### DISCUSSION TITLE
Generating hooks from the non-react version of `createApi`

### DISCUSSION THREAD

**Discussion by DanCouper** [reduxjs/redux-toolkit#2067] [Q&A]
I've got a RTK/RTK Query setup that I'm putting together, and [as things stand] while I build out the structure + tests it's not connected to React at all. I want to keep it this way: the end result will be a standalone configurable package (essentially a domain logic layer) I can drop into a number of different applications I work on. These are all currently React/React Native, but I want to put together some PoC/tests using other frameworks, so it's useful for me to have no hard dependency on React.

I'm digging around the `query` codebase and before I go down the rabbit hole and end up with some hacked together solution, is there a simple way to take the query code constructed using `createApi` and, within a given app (or as a seperate optional module/export in my package), enhance it so that I can autogenerate the React hooks? *ie* get the automatic hook generation behaviour, but only when I need it?

**Comment by phryneas** [comment]
Nope, I'm sorry.
